src/runtime/methodbinder.cs

```diff
@@ -335,6 +356,58 @@ internal Binding Bind(IntPtr inst, IntPtr args, IntPtr kw, MethodBase info, Meth
                     continue;
                 }
 
+                var matchedMethod = new MatchedMethod(kwargsMatched, defaultsNeeded, margs, outs, mi);
+                argMatchedMethods.Add(matchedMethod);
+            }
+            if (argMatchedMethods.Count() > 0)
+            {
+                // Order matched methods by number of kwargs matched and get the max possible number
+                // of kwargs matched
+                var bestKwargMatchCount = argMatchedMethods.Max(x => x.KwargsMatched);
+
+                List<MatchedMethod> bestKwargMatches = new List<MatchedMethod>(argMatchedMethods.Count());
+                foreach (MatchedMethod testMatch in argMatchedMethods)
+                {
+                    if (testMatch.KwargsMatched == bestKwargMatchCount)
+                    {
+                        bestKwargMatches.Add(testMatch);
+                    }
+                }
+
+                // Order by the number of defaults required and find the smallest
+                var fewestDefaultsRequired  = bestKwargMatches.Min(x => x.DefaultsNeeded);
+                int bestCount = 0;
+                int bestMatchIndex = -1;
+
+                foreach (MatchedMethod testMatch in bestKwargMatches)
+                {
+                    if (testMatch.DefaultsNeeded == fewestDefaultsRequired)
+                    {
+                        bestCount++;
+                        if (bestMatchIndex == -1)
```

Comment:
Not having this check will cause tests to fail because a different overload is picked- e.g. MethodTest.TestOverloadedObjectTwo(5, 5) will get the (Object, Object) overload rather than the (Int, Int) overload